src/test/java/api/loans/OverrideCheckOutByBarcodeTests.java

```diff
@@ -326,8 +354,191 @@ public void canCreateRecallRequestAfterOverriddenCheckout()
+      hasParameter("loanPolicyName", "Not Loanable Policy"))));
+  }
+
+  @Test
+  public void cannotOverrideItemNotLoanableBlockWhenUserDoesNotHavePermissions() {
+    setNotLoanablePolicy();
+    Response response = checkOutFixture.attemptCheckOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(itemsFixture.basedUponSmallAngryPlanet())
+        .to(usersFixture.steve())
+        .at(UUID.randomUUID())
+        .on(TEST_LOAN_DATE).withOverrideBlocks(new OverrideBlocks(
+          new ItemNotLoanableBlock(TEST_DUE_DATE), null, null, TEST_COMMENT)));
+
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getJson(), hasErrorWith(allOf(
+      hasMessage(INSUFFICIENT_OVERRIDE_PERMISSIONS))));
+    assertThat(getMissingPermissions(response), hasSize(1));
+    assertThat(getMissingPermissions(response), hasItem(OVERRIDE_ITEM_NOT_LOANABLE_BLOCK_PERMISSION));
+  }
+
+  @Test
+  public void cannotOverrideItemNotLoanableBlockWhenUserDoesNotHaveRequiredPermissions() {
+    final OkapiHeaders okapiHeaders = buildOkapiHeadersWithPermissions(
+      OVERRIDE_PATRON_BLOCK_PERMISSION);
+
+    setNotLoanablePolicy();
+    Response response = checkOutFixture.attemptCheckOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(itemsFixture.basedUponSmallAngryPlanet())
+        .to(usersFixture.steve())
+        .at(UUID.randomUUID())
+        .on(TEST_LOAN_DATE).withOverrideBlocks(new OverrideBlocks(
+        new ItemNotLoanableBlock(TEST_DUE_DATE), null, null, TEST_COMMENT)),
+      okapiHeaders);
+
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getJson(), hasErrorWith(allOf(
+      hasMessage(INSUFFICIENT_OVERRIDE_PERMISSIONS))));
+    assertThat(getMissingPermissions(response), hasSize(1));
+    assertThat(getMissingPermissions(response), hasItem(OVERRIDE_ITEM_NOT_LOANABLE_BLOCK_PERMISSION));
+  }
+
+  @Test
+  public void cannotOverrideItemNotLoanableBlockAndPatronBlockWhenUserDoesNotHavePermissions() {
+    setNotLoanablePolicy();
+    Response response = checkOutFixture.attemptCheckOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(itemsFixture.basedUponSmallAngryPlanet())
+        .to(usersFixture.steve())
+        .at(UUID.randomUUID())
+        .on(TEST_LOAN_DATE).withOverrideBlocks(new OverrideBlocks(
+        new ItemNotLoanableBlock(TEST_DUE_DATE), new PatronBlock(), null, TEST_COMMENT)));
+
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getJson(), hasErrorWith(allOf(
+      hasMessage(INSUFFICIENT_OVERRIDE_PERMISSIONS))));
+    assertThat(getMissingPermissions(response), hasSize(2));
+    assertThat(getMissingPermissions(response).get(0), is(OVERRIDE_PATRON_BLOCK_PERMISSION));
+    assertThat(getMissingPermissions(response).get(1), is(OVERRIDE_ITEM_NOT_LOANABLE_BLOCK_PERMISSION));
+  }
+
+  @Test
+  public void canOverrideCheckoutWhenItemLimitWasReachedForBookMaterialType() {
```

Comment:
Can we cover successful override scenarios for patron block and item not loanable block?